assets/src/components/detours/detourRouteSelectionPanel.tsx

```diff
@@ -0,0 +1,92 @@
+import React from "react"
+import { Button, FormSelect } from "react-bootstrap"
+import { Panel } from "./diversionPage"
+import {
+  ByRoutePatternId,
+  Route,
+  RoutePattern,
+  RoutePatternId,
+} from "../../schedule"
+import RoutePropertiesCard from "../mapPage/routePropertiesCard"
+
+interface SelectedRouteInfoWithRoute {
+  selectedRoute: Route
+  routePatterns: ByRoutePatternId<RoutePattern>
+  selectedRoutePatternId: RoutePatternId | null
+}
+
+interface SelectedRouteInfoWithoutRoute {
+  selectedRoute: null
+}
+
+type SelectedRouteInfo =
+  | SelectedRouteInfoWithRoute
+  | SelectedRouteInfoWithoutRoute
+
+interface DetourRouteSelectionPanelProps {
+  allRoutes: Route[]
+  selectedRouteInfo: SelectedRouteInfo
+}
+
+const selectedRoutePatternForInfo = (
```

Comment:
super minor non-blocking nit: "for" sounds weird to me and I'd _personally_ prefer `From` but don't feel pressured to.
```suggestion
const selectedRoutePatternFromInfo = (
```

Reply:
[I also like that better](https://github.com/mbta/skate/pull/2614/commits/f5fb48751652ed56039441d4700a82ae5a774f33)